compiler/package.yaml

```diff
@@ -103,3 +117,13 @@ tests:
     - shakespeare
     - megaparsec
     - quickcheck-arbitrary-template
+    when:
+      - condition: os(windows) && arch(i386)
+        ld-options:     -Wl,--dynamicbase -Wl,--nxcompat -Wl,--large-address-aware
+      - condition: os(windows) && arch(x86_64)
+        ld-options:     -Wl,--dynamicbase -Wl,--nxcompat -Wl,--high-entropy-va
+      - condition: flag(warningsAsError)
+        then:
+          ghc-options:    -threaded -Wall -Werror -rtsopts -with-rtsopts=-N
```

Comment:
Minor: if we put the extra bits at the end, then it's easier to keep the two lines in sync:

```
  then:
    ghc-options: -threaded -rtsopts -with-rtsopts=-N -Wall -Werror
  else:
    ghc-options: -threaded -rtsopts -with-rtsopts=-N
```